src/main/java/fr/openent/crre/service/impl/DefaultNotificationService.java

```diff
@@ -0,0 +1,222 @@
+                        if (projectModelListMapAll.keySet().stream().anyMatch(projectModel1 -> projectModel1.getStructureId().equals(projectModel.getStructureId()))){
+                            projectModelListMapAll.get(projectModelListMapAll.keySet().stream().filter(projectModel1 -> projectModel1.getStructureId().equals(projectModel.getStructureId())).findFirst().orElse(null)).addAll(orderRegionEquipmentModels);
+                        } else {
+                            projectModelListMapAll.put(projectModel, orderRegionEquipmentModels);
+                        }
+                    });
+                    List<String> structureIdList = projectModelListMapAll.entrySet().stream()
+                    .filter(this::isProjectIsComplete)
+                    .map(projectModelListEntry -> projectModelListEntry.getKey().getStructureId())
+                    .distinct()
+                    .collect(Collectors.toList());
+
+                    return this.serviceFactory.getUserService().getValidatorUser(structureIdList);
+                })
+                .onSuccess(userStructureMap -> {
+                    final Map<String, List<OrderRegionEquipmentModel>> userIdOrderRegionMap = userStructureMap.entrySet().stream()
+                            .collect(Collectors.toMap(neo4jUserModelStringEntry -> neo4jUserModelStringEntry.getKey().getData().getId(),
+                                    neo4jUserModelStringEntry -> projectModelListMapAll.entrySet().stream()
+                                            .filter(this::isProjectIsComplete)
+                                            .filter(projectModelListEntry -> neo4jUserModelStringEntry.getValue().equals(projectModelListEntry.getKey().getStructureId()))
+                                            .findFirst()
+                                            .map(Map.Entry::getValue)
+                                            .orElse(new ArrayList<>())));
+
+                    userIdOrderRegionMap.forEach(this::prepareMessageToValidator);
+                })
+                .onFailure(error -> log.error(String.format("[CRRE@%s::sendNotificationToValidator] Fail to send notification to validator %s",
+                        this.getClass().getSimpleName(), error.getMessage())));
+    }
+
+    private void prepareMessageToValidator(String userId, List<OrderRegionEquipmentModel> orderRegionEquipmentModels) {
+        if (orderRegionEquipmentModels.isEmpty()) {
+            return;
+        }
+
+        String i18nStatus = this.getI18nStatusToValidator(orderRegionEquipmentModels);
+        UserUtils.getUserInfos(this.eventBus, userId, userInfos -> {
+            String local = null;
+            try {
+                local = new JsonObject((String) ((LinkedHashMap<?, ?>) userInfos.getAttribute(Field.PREFERENCES)).get(Field.LANGUAGE)).getString(Field.DEFAULT_DASH_DOMAIN);
+            } catch (Exception ignored) {
+            }
```

Comment:
Je ne comprend pas ici le faite de laisser un catch vide.

Reply:
S'il y a une erreur je n'ai pas besoin de faire autre chose. Je peux faire ça mais cela reviens au meme
```suggestion
            String local;
            try {
                local = new JsonObject((String) ((LinkedHashMap<?, ?>) userInfos.getAttribute(Field.PREFERENCES)).get(Field.LANGUAGE)).getString(Field.DEFAULT_DASH_DOMAIN);
            } catch (Exception ignored) {
                local = null;
            }
```

Reply:
tu ne peux pas enlever le try du coup ?

Reply:
non car s'il y a un exception telque un CastException, oon ne va pas envoyer la notification